blockindex/bloomfilterindexer.go

```diff
@@ -206,24 +211,26 @@ func (bfx *bloomfilterIndexer) blockBloomFilter(blockNumber uint64) (bloom.Bloom
 
 func (bfx *bloomfilterIndexer) delete(blockNumber uint64) error {
 	rangeBloomfilterKey := bfx.rangeBloomfilterKey(blockNumber)
-	if err := bfx.kvStore.Delete(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey)); err != nil {
-		return errors.Wrapf(err, "failed to delete RangeBloomFilter")
-	}
-	if err := bfx.kvStore.Delete(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber)); err != nil {
-		return errors.Wrapf(err, "failed to delete BlockBloomFilter")
+	bfx.flusher.KVStoreWithBuffer().MustDelete(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey))
+	bfx.flusher.KVStoreWithBuffer().MustDelete(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber))
+	bfx.flusher.KVStoreWithBuffer().MustDelete(RangeBloomFilterNamespace, []byte(CurrentHeightKey))
```

Comment:
use "Delete"

Reply:
anyway "Delete" always return nil and behaves the same as "MustDelete" - why do "Delete" is better?

Reply:
addressed on the latest commit!